ClosedXML/Excel/Drawings/XLMarker.cs

```diff
@@ -1,23 +1,38 @@
+// Keep this file CodeMaid organised and cleaned
+using System;
 using System.Diagnostics;
 using System.Drawing;
 
 namespace ClosedXML.Excel.Drawings
 {
     [DebuggerDisplay("{Address} {Offset}")]
-    internal class XLMarker : IXLMarker
+    internal class XLMarker
     {
-        internal XLMarker(IXLAddress address)
-            : this(address, new Point(0, 0))
+        // Using a range to store the lcoation so that it gets added to the range repository
+        // and hence will be adjusted when there are insertions / deletions
+        private readonly IXLRange rangeCell;
+
+        internal XLMarker(IXLCell cell)
+                    : this(cell.AsRange(), new Point(0, 0))
+        { }
+
+        internal XLMarker(IXLRange rangeCell)
+            : this(rangeCell, new Point(0, 0))
         { }
 
-        internal XLMarker(IXLAddress address, Point offset)
+        internal XLMarker(IXLCell cell, Point offset)
+            : this(cell.AsRange(), offset)
+        { }
+
+        internal XLMarker(IXLRange rangeCell, Point offset)
         {
-            this.Address = address;
+            this.rangeCell = rangeCell;
```

Comment:
I'd suggest adding the validation that `rangeCell` is actually one cell high and one cell wide and throw the `ArgumentException` otherwise. Thus we'll protect ourselves from misusage. Besides, markers constructed for ranges `A1:A1` and `A1:B2` functionally would be equal, but their `rangeCell` would differ

Reply:
True, I'm going to change the constructor modifier too. Should be `private`.